src/components/ui/input/index.js

```diff
@@ -0,0 +1,69 @@
+import React, {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from 'react';
+
+import './input.css';
+
+const MIN_HEIGHT = 20;
+
+const Input = ({
+  characterLimit = false,
```

Comment:
if we set this to 0 by default then we would not need the seperate check for `isLimitSet`
```suggestion
  characterLimit = 0,
```